cpp/src/io/json/json_column.cu

```diff
@@ -530,13 +530,15 @@ void make_device_json_column(device_span<SymbolT const> input,
   auto max_row_offsets = cudf::detail::make_std_vector_async(d_max_row_offsets, stream);
   std::vector<std::string> column_names = copy_strings_to_host(
     input, d_column_tree.node_range_begin, d_column_tree.node_range_end, stream);
+  stream.synchronize();
```

Comment:
Why? `copy_strings_to_host` sync inside it.

Reply:
IMO, we should rename `copy_strings_to_host`  into `copy_strings_to_host_sync` to explicitly tell that we have a sync there.

Reply:
Good suggestion. I was thinking that too.
We need sync because it's copied to host and it will be used later as column names in host.